client/cmd/dexc/ui/widgets.go

```diff
@@ -0,0 +1,464 @@
+	"decred.org/dcrdex/client/rpcserver"
+	"decred.org/dcrdex/client/webserver"
+	"github.com/decred/slog"
+	"github.com/gdamore/tcell"
+	"github.com/rivo/tview"
+)
+
+var (
+	// The application context. Provided to the Run function, but stored globally.
+	appCtx context.Context
+	// The tview application.
+	app *tview.Application
+	// The core DEX client application. Used by both the RPC server and the
+	// web server.
+	clientCore *core.Core
+	// These are the main view widgets and the loggers attached to their journals.
+	screen            *Screen
+	mainMenu          *chooser
+	appJournal        *journal
+	webView           *serverView
+	rpcView           *serverView
+	marketView        *marketViewer
+	acctsView         *accountsViewer
+	noteJournal       *journal
+	noteLog           slog.Logger
+	marketLog         slog.Logger
+	acctsLog          slog.Logger
+	focusColor        = tcell.GetColor("#dedeff")
+	blurColor         = tcell.GetColor("grey")
+	onColor           = tcell.GetColor("green")
+	offColor          = blurColor
+	colorBlack        = tcell.GetColor("black")
+	notificationCount uint32
+)
+
+// Alias for brevity. A commonly used tview callback.
+type inputCapture func(event *tcell.EventKey) *tcell.EventKey
+
+// Run the TUI app.
+func Run(ctx context.Context) {
+	appCtx = ctx
+	// Initialize logging to the main application log view.
+	appJournal = newJournal("Application Log", handleAppLogKey)
+	InitLogging(func(p []byte) {
+		appJournal.Write(p)
+	})
+	// Close closes the log rotator.
+	defer Close()
+	// Create the UI and start the app.
+	createApp()
+	if err := app.SetRoot(screen, true).SetFocus(mainMenu).Run(); err != nil {
+		panic(err)
+	}
+}
+
+// A focuser is satisfied by anything that embeds *tview.Box and implements
+// AddFocus and RemoveFocus methods. The two additional methods are not from
+// tview, and are used to help with focus control and chaining.
+type focuser interface {
+	tview.Primitive
+	SetInputCapture(capture func(event *tcell.EventKey) *tcell.EventKey) *tview.Box
+	GetInputCapture() func(event *tcell.EventKey) *tcell.EventKey
+	// AddFocus and RemoveFocus enable additional control over navigation.
+	AddFocus()
+	RemoveFocus()
+}
+
+// Screen is the the full screen. It is separated into a permanent menu on the
+// left, and a swappable view on the right.
+type Screen struct {
+	*tview.Flex
+	right   focuser
+	focused focuser
+}
+
+var welcomeMessage = "Welcome to Decred DEX. Use [#838ac7]Up[white] and " +
+	"[#838ac7]Down[white] arrows and then press [#838ac7]Enter[white] to select" +
+	" a new view. The [#838ac7]Escape[white] key will usually toggle the focus" +
+	" between the menu and the currently selected view. When the selected view" +
+	" is focused, most navigation can be done with your the [#838ac7]Left" +
+	"[white] and [#838ac7]Right[white] arrows, or alternatively [#838ac7]Tab" +
+	"[white] and [#838ac7]Shift+Tab[white]. Use [#838ac7]Escape[white] to remove" +
+	" focus from a form element."
```

Comment:
The white on grey is hard for me to see...

Reply:
White on grey? Do you mean the purple parts? The white text with default background is default console coloring, I think.

Reply:
If you look at the first pic I uploaded you can see what I see by default.

Reply:
Oh weird. I'll look into it.

Reply:
https://github.com/decred/dcrdex/blob/47ece4c4c4379a154bce2d90ce8fb26fc27d2042/client/cmd/dexc/ui/widgets.go#L473

See if that works for you when you get a chance.

Reply:
No, no visible difference. I have setterm --inversescreen on and tried turning that off, and while the background turned black, I couldn't see the parts that currently look white at all.  It may just be some funky settings on my side, but I don't think I am setting anything besides inversescreen.
Can I see a screenshot from you? What are the colors supposed to be like?

Reply:
Ok! so, running like so `xterm ./dexc` and I get a nice picture. Creating a new xterm window for the app has fixed the colors for me.
<details close>
<summary>pic</summary>

![colors](https://user-images.githubusercontent.com/34998433/72025395-597f2780-32bb-11ea-8dc8-03a3178f327a.png)


</details>

Reply:
What interesting is that while the inline-colored `(1)` next to `Notifications` is colored yellow instead of orange, and the inline key specifier text in the application log text is not purple. So it looks like some color incompatibility problems in general. Maybe you could peek around once this is merged and see if you can find better color values.

![image](https://user-images.githubusercontent.com/6109680/72026624-b8a05600-3241-11ea-959f-8194df8991cf.png)

Reply:
Yeah, the purple isn't showing through at all for me. Are you using xterm?

Reply:
Also, I didn't realize it was purple, that's another constant you could make...

Reply:
The color incompatibility is especially odd consider **tview** is based on **tcell**, who's readme opens with

> Tcell is a Go package that provides a cell based view for text terminals, **like xterm**. It was inspired by termbox, but includes many additional improvements.